### DISCUSSION TITLE
Error while adding aliases to KnowledgeBase while training Entity Linker

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10924] [Help: Coding & Implementations] [answered]
Hi!I am training a custom Entity Linker component and creating a Knowledge base for the same using the following code:
```
for a, qids in alias_dict.items():
        kb.add_alias(alias=a,entities=qids,probabilities=[0.5])
```
where `alias_dict` is a dictionary with aliases as keys and a list of entities for each alias as keys. For eg, if entities `B` and `C` have aliases `'a'`, then
`alias_dict['a']=['b','c']`

Now I get the following error while trying to add them:
`ValueError: [E132] The vectors for entities and probabilities for alias  should have equal length, but found 2 and 1 respectively.`
What does this mean exactly?Does it mean that both entities have same vectors?

**Comment by polm** [comment, marked as answer]
The `entities` argument must be the same length as the the `probabilities` argument. So if your `qids` has a length of two then your `probabilities` should be `[0.5, 0.5]` or something, not just `[0.5]`.

**Reply by shrinidhin** [reply]
Oh I see what I did wrong!Thank you!
